src/utils.ts

```diff
@@ -118,4 +120,49 @@ export class Utils {
             throw new TypeError(`${name}: Expected number but got ${typeof value}`);
         }
     }
+
+    /**
+     * @hidden
+     * Converts a string in the format `[-][d.]hh:mm:ss[.fffffff]`
+     * to a `Duration` object
+     * (This is the format the default .NET serializer uses, and the
+     * format coming from the Durable extension)
+     * @param input the formatted string
+     */
+    public static durationFromString(input: string): Duration {

```

Comment:
This was necessary because .NET serializes `TimeSpan` objects in this weird, non-ISO standard format: `[-][d.]hh:mm:ss[.fffffff]`. See [here](https://learn.microsoft.com/en-us/dotnet/api/system.timespan.tostring?view=net-7.0) for more details. This is the format of the durations the SDK receives in its payload from the Extension. `moment.js` seemed to handle this fine out of the gate, but I couldn't find a `luxon` method that either accepts this format, or even allows specifying a custom format. The same is also true for `dayjs`, the other contender for the replacement. I know it's not ideal, but I also added unit tests to make sure this logic is sound, and step-by-step comments. If anyone has better suggestions, I'm happy to hear them.  There is a GH issue here: https://github.com/moment/luxon/issues/296 on luxon to add a `Duration.fromFormat()` method, which would help us in this case, but until that's closed, this is the best I could come up with :)

Reply:
Can we punt this PR until after preview? I believe we discussed this during triage and determined the durable package doesn't export any moment types in its public API (right?), so it's more of an internal engineering change as opposed to something that will directly benefit the user.

Even after preview, I think the fact that we have to manually implement `durationFromString` is reason enough to stay on moment. It's not like the package is dead - it's still in maintenance mode.

Reply:
The Durable SDK does list `moment` as one of its dependencies, and it _does_ export `moment` types in its public types, as the type of some private fields, which are included in the auto-generated `.d.ts` files by default.

Reply:
@hossam-nasr: just to make sure I understand the context. If we're moving to manually generated types, would we be able to avoid exporting `moment`?

Reply:
@davidmrdavid Yes, I believe so. But it will stay in our dependencies, even if it's not in our public types (and therefore changing it in the future won't be a breaking change). We can def still do this after preview, but I thought we agreed during triage to just do it now if the fix is easy enough.

Reply:
> it's not in our public types (and therefore changing it in the future won't be a breaking change)

Given this, I think we should stay on moment until Luxon or dayjs add support for the format we need